src/high-level/src/KinDynComputations.cpp

```diff
@@ -1073,6 +1073,26 @@ Transform KinDynComputations::getWorldTransform(const FrameIndex frameIndex)
     return world_H_frame;
 }
 
+std::vector<iDynTree::Matrix4x4> KinDynComputations::getWorldTransformsAsHomogeneous(std::vector<std::string> frameNames)
+{
+    std::vector<iDynTree::Matrix4x4> worldTransforms;
+    int numberOfTransforms=frameNames.size();
+    for(int number=0;number<numberOfTransforms;number++)
+    {
+        std::string frameName=frameNames[number];
+        int frameIndex = getFrameIndex(frameName);
+        if( frameIndex < 0 )
+        {
+            worldTransforms.push_back(Transform::Identity().asHomogeneousTransform());
+        }
```

Comment:
Silently failing is bad, because if a user puts a typo in the input frames, it should get a clear error, not the transform silently being returned as identity. If there is an error in the input data, I suggest to print a clear error and signal the error in the return value in some way (for example returning the empty vector could  be a way to signal the error).

Reply:
I was thinking about that. I had done it in one of the functions I erased I think. I will add the reporterror. I was unsure if I want it to fail all together or just a warning.

Reply:
Now that I take another look at it,  getFrameIndex should show an error if the frame is not there. Is that not enough?

Reply:
I think it is still better to print something to have a sort of stack trace of the error.

Reply:
Ok Then again I will probably repeat the same message but from the other function.

Reply:
Otherwise, you can run a check first with [``isFrameNameUsed()``](http://wiki.icub.org/codyco/dox/html/idyntree/html/classiDynTree_1_1Model.html#a6da12797ce7fd9bc747a6192e1a8632d) so that you avoid the other method to print the error before you.